dvc/stage/monitor.py

```diff
@@ -75,6 +75,7 @@ def __init__(
     def _run_callback(stage, callback_func):
         stage.save(allow_missing=True)
         stage.commit(allow_missing=True)
+        stage.unprotect_outs()
```

Comment:
A better approach here would be to have a method that could just save the outputs to cache without touching the originals, kinda like our `--to-cache/--to-remote` stuff. That would save us some unnecessary moving+linking+copying that we have right now and would turn that into just `copying`. Looks like it might be a special option for `commit` that would tell it to only `copy`. Might be okay to leave it as is for now though.